### ISSUE TITLE
[BUG]How can I make the bot ignore the invalid UTF-8 response?

### ISSUE THREAD

**Issue by rikonaka** [python-telegram-bot/python-telegram-bot#1609]
### Steps to reproduce
1. I think it is a small probability event, so I don't know how to reproduce it.

### Expected behaviour
Run perfectly.

### Actual behaviour
Bot raise error.
```
2019-11-15 12:50:38,048 - telegram.ext.updater - ERROR - Error while getting Updates: Server response could not be decoded using UTF-8
2019-11-15 12:50:38,048 - xxbot.__main__ - WARNING - Update None caused error Server response could not be decoded using UTF-8
2019-11-15 12:50:39,220 - telegram.ext.updater - ERROR - Error while getting Updates: Server response could not be decoded using UTF-8
2019-11-15 12:50:39,221 - xxbot.__main__ - WARNING - Update None caused error Server response could not be decoded using UTF-8
```
Those error message has been constantly appearing and repeating, causing the program to not work properly.

And I change the `logging` module form `INFO` to `DEBUG`, the more error details show like this.
```
2019-11-15 11:02:44,710 - telegram.ext.dispatcher - DEBUG - Setting singleton dispatcher as <telegram.ext.dispatcher.Dispatcher object at 0x7ff7b973f358>
2019-11-15 11:02:44,714 - JobQueue - DEBUG - Putting job repeat_submission with t=1573815764.714072
2019-11-15 11:02:44,714 - JobQueue - DEBUG - Putting job repeat_submission_list_clear with t=1573815764.714721
2019-11-15 11:02:44,714 - JobQueue - DEBUG - Putting job repeat_table_name with t=1573815764.714840
2019-11-15 11:02:44,714 - JobQueue - DEBUG - Putting job repeat_table_name_ugly with t=1573815764.714952
2019-11-15 11:02:44,715 - JobQueue - DEBUG - Putting job repeat_database_process with t=1573815764.715042
2019-11-15 11:02:44,715 - JobQueue - DEBUG - Putting job repeat_repeat_list_delete_one with t=1573815764.715124
2019-11-15 11:02:44,715 - JobQueue - DEBUG - Putting job repeat_appreciate_list_delete_one with t=1573815764.715253
2019-11-15 11:02:44,715 - JobQueue - DEBUG - Putting job repeat_rest_list_delete_one with t=1573815764.715346
2019-11-15 11:02:44,715 - JobQueue - DEBUG - Putting job repeat_sort_by_date_list_delete_one with t=1573815764.715449
2019-11-15 11:02:44,715 - JobQueue - DEBUG - Putting job repeat_pic_pool_complete with t=1573815764.715537
2019-11-15 11:02:44,715 - JobQueue - DEBUG - Putting job repeat_pic_pool_check with t=1573815764.715622
2019-11-15 11:02:44,715 - JobQueue - DEBUG - Putting job repeat_non_pic_pool_complete with t=1573815764.715753
2019-11-15 11:02:44,715 - JobQueue - DEBUG - Putting job repeat_non_pic_pool_check with t=1573815764.715847
2019-11-15 11:02:44,716 - JobQueue - DEBUG - Ticking jobs with t=1573815764.716447
2019-11-15 11:02:44,716 - JobQueue - DEBUG - JobQueue thread started
2019-11-15 11:02:44,716 - JobQueue - DEBUG - Peeked at repeat_submission with t=1573815764.714072
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Running job repeat_submission
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Putting job repeat_submission with t=1573815770.714072
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Peeked at repeat_submission_list_clear with t=1573815764.714721
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Running job repeat_submission_list_clear
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Putting job repeat_submission_list_clear with t=1573816124.714721
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Peeked at repeat_table_name with t=1573815764.714840
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Running job repeat_table_name
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Putting job repeat_table_name with t=1573815770.714840
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Peeked at repeat_table_name_ugly with t=1573815764.714952
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Running job repeat_table_name_ugly
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Putting job repeat_table_name_ugly with t=1573815770.714952
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Peeked at repeat_database_process with t=1573815764.715042
2019-11-15 11:02:44,717 - JobQueue - DEBUG - Running job repeat_database_process
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Putting job repeat_database_process with t=1573815767.715042
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Peeked at repeat_repeat_list_delete_one with t=1573815764.715124
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Running job repeat_repeat_list_delete_one
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Putting job repeat_repeat_list_delete_one with t=1573816364.715124
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Peeked at repeat_appreciate_list_delete_one with t=1573815764.715253
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Running job repeat_appreciate_list_delete_one
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Putting job repeat_appreciate_list_delete_one with t=1573815824.715253
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Peeked at repeat_rest_list_delete_one with t=1573815764.715346
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Running job repeat_rest_list_delete_one
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Putting job repeat_rest_list_delete_one with t=1573815824.715346
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Peeked at repeat_sort_by_date_list_delete_one with t=1573815764.715449
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Running job repeat_sort_by_date_list_delete_one
2019-11-15 11:02:44,718 - JobQueue - DEBUG - Putting job repeat_sort_by_date_list_delete_one with t=1573816364.715449
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Peeked at repeat_pic_pool_complete with t=1573815764.715537
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Running job repeat_pic_pool_complete
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Putting job repeat_pic_pool_complete with t=1573815824.715537
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Peeked at repeat_pic_pool_check with t=1573815764.715622
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Running job repeat_pic_pool_check
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Putting job repeat_pic_pool_check with t=1573815765.715622
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Peeked at repeat_non_pic_pool_complete with t=1573815764.715753
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Running job repeat_non_pic_pool_complete
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Putting job repeat_non_pic_pool_complete with t=1573815824.715753
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Peeked at repeat_non_pic_pool_check with t=1573815764.715847
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Running job repeat_non_pic_pool_check
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Putting job repeat_non_pic_pool_check with t=1573815765.715847
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Peeked at repeat_pic_pool_check with t=1573815765.715622
2019-11-15 11:02:44,719 - JobQueue - DEBUG - Next task isn't due yet. Finished!
2019-11-15 11:02:44,720 - telegram.ext.updater - DEBUG - dispatcher - started
2019-11-15 11:02:44,723 - telegram.ext.updater - DEBUG - updater - started
2019-11-15 11:02:44,723 - telegram.ext.updater - DEBUG - Updater thread started (polling)
2019-11-15 11:02:44,723 - telegram.ext.updater - DEBUG - Start network loop retry bootstrap del webhook
2019-11-15 11:02:44,724 - telegram.bot - DEBUG - Entering: delete_webhook
2019-11-15 11:02:44,724 - telegram.vendor.ptb_urllib3.urllib3.connectionpool - DEBUG - Starting new HTTPS connection (1): api.telegram.org
2019-11-15 11:02:44,787 - telegram.ext.dispatcher - DEBUG - Dispatcher started
2019-11-15 11:02:45,435 - telegram.vendor.ptb_urllib3.urllib3.connectionpool - DEBUG - https://api.telegram.org:443 "POST /bot627323681:AAHBB-8QBgyu7wPhR_GHIrbbH40YjT5kB2I/deleteWebhook HTTP/1.1" 200 68
2019-11-15 11:02:45,435 - telegram.bot - DEBUG - True
2019-11-15 11:02:45,435 - telegram.bot - DEBUG - Exiting: delete_webhook
2019-11-15 11:02:45,435 - telegram.ext.updater - DEBUG - Bootstrap done
2019-11-15 11:02:45,435 - telegram.ext.updater - DEBUG - Start network loop retry getting Updates
2019-11-15 11:02:45,436 - telegram.bot - DEBUG - Entering: get_updates
2019-11-15 11:02:45,593 - telegram.vendor.ptb_urllib3.urllib3.connectionpool - DEBUG - https://api.telegram.org:443 "POST /bot627323681:AAHBB-8QBgyu7wPhR_GHIrbbH40YjT5kB2I/getUpdates HTTP/1.1" 200 5909
2019-11-15 11:02:45,593 - telegram.utils.request - DEBUG - Logging raw invalid UTF-8 response:
b'{"ok":true,"result":[{"update_id":XXXXXXXXX,\n"message":{"message_id":79819,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","last_name":"XXXXXXXXX","username":"XXXXXXXXX"},"chat":{"id":-XXXXXXXXX,"title":"XXXXXXXXX","username":"XXXXXXXXX","type":"supergroup"},"date":1573803803,"reply_to_message":{"message_id":79818,"from":{"id":XXXXXXXXX,"is_bot":true,"first_name":"XXXXXXXXX","username":"XXXXXXXXX"},"chat":{"id":-XXXXXXXXX,"title":"XXXXXXXXX","username":"XXXXXXXXX","type":"supergroup"},"date":1573803744,"photo":[{"file_id":"XXXXXXXXX","file_size":3620,"width":200,"height":70}],"caption":"XXXXXXXXX","caption_entities":[{"offset":3,"length":4,"type":"text_mention","user":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","last_name":"XXXXXXXXX","username":"XXXXXXXXX"}}],"reply_markup":{"inline_keyboard":[[{"text":"XXXXXXXXX","callback_data":"XXXXXXXXX"}],[{"text":"XXXXXXXXX","callback_data":"XXXXXXXXX"},{"text":"XXXXXXXXX","callback_data":"XXXXXXXXX"}]]}},"text":"XXXXXXXXX"}},{"update_id":XXXXXXXXX,\n"message":{"message_id":1807,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX"},"chat":{"id":-XXXXXXXXX,"title":"XXXXXXXXX","username":"XXXXXXXXX","type":"supergroup"},"date":1573804787,"new_chat_participant":{"id":XXXXXXXXX,"is_bot":false,"first_name":" XXXXXXXXX"},"new_chat_member":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX"},"new_chat_members":[{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX"}]}},{"update_id":XXXXXXXXX,\n"message":{"message_id":129646,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","language_code":"zh-hans"},"chat":{"id":-XXXXXXXXX,"title":"XXXXXXXXX","username":"XXXXXXXXX","type":"supergroup"},"date":1573805383,"text":"XXXXXXXXX","entities":[{"offset":0,"length":5,"type":"bot_command"}]}},{"update_id":XXXXXXXXX,\n"message":{"message_id":129647,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","language_code":"zh-hans"},"chat":{"id":-XXXXXXXXX,"title":"XXXXXXXXX","username":"XXXXXXXXX","type":"supergroup"},"date":1573805420,"text":"XXXXXXXXX","entities":[{"offset":0,"length":2,"type":"bot_command"}]}},{"update_id":XXXXXXXXX,\n"message":{"message_id":129648,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX","language_code":"en"},"chat":{"id":-XXXXXXXXX,"title":"XXXXXXXXX","username":"XXXXXXXXX","type":"supergroup"},"date":1573810865,"text":"XXXXXXXXX","entities":[{"offset":0,"length":24,"type":"bot_command"}]}},{"update_id":XXXXXXXXX,\n"message":{"message_id":129649,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX","language_code":"en"},"chat":{"id":-XXXXXXXXX,"title":"XXXXXXXXX","username":"XXXXXXXXX","type":"supergroup"},"date":1573810886,"text":"XXXXXXXXX","entities":[{"offset":0,"length":24,"type":"bot_command"}]}},{"update_id":XXXXXXXXX,\n"message":{"message_id":129650,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX","language_code":"en"},"chat":{"id":-XXXXXXXXX,"title":"XXXXXXXXX","username":"XXXXXXXXX","type":"supergroup"},"date":1573810890,"text":"XXXXXXXXX","entities":[{"offset":0,"length":17,"type":"bot_command"}]}},{"update_id":XXXXXXXXX,\n"message":{"message_id":109851,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX","language_code":"en"},"chat":{"id":XXXXXXXXX,"first_name":"XXXXXXXXX","username":"XXXXXXXXX","type":"private"},"date":1573811048,"text":"XXXXXXXXX","entities":[{"offset":0,"length":12,"type":"bot_command"}]}},{"update_id":XXXXXXXXX,\n"message":{"message_id":XXXXXXXXX,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX","language_code":"en"},"chat":{"id":366039180,"first_name":"XXXXXXXXX","username":"XXXXXXXXX","type":"private"},"date":1573811339,"text":"XXXXXXXXX","entities":[{"offset":0,"length":12,"type":"bot_command"}]}},{"update_id":XXXXXXXXX,\n"message":{"message_id":XXXXXXXXX,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX","language_code":"en"},"chat":{"id":XXXXXXXXX,"first_name":"XXXXXXXXX","username":"XXXXXXXXX","type":"private"},"date":1573811393,"text":"XXXXXXXXX","entities":[{"offset":0,"length":12,"type":"bot_command"}]}},{"update_id":XXXXXXXXX,\n"message":{"message_id":129651,"from":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX"},"chat":{"id":-XXXXXXXXX,"title":"XXXXXXXXX","username":"XXXXXXXXX","type":"supergroup"},"date":1573813538,"new_chat_participant":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX"},"new_chat_member":{"id":XXXXXXXXX,"is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX"},"new_chat_members":[{"id":XXXXXXXXX,is_bot":false,"first_name":"XXXXXXXXX","username":"XXXXXXXXX"}]}}]}'
2019-11-15 11:02:45,593 - telegram.ext.updater - ERROR - Error while getting Updates: Server response could not be decoded using UTF-8
2019-11-15 11:02:45,593 - telegram.ext.dispatcher - DEBUG - Processing Update: Server response could not be decoded using UTF-8
2019-11-15 11:02:45,594 - xxbot.__main__ - WARNING - Update None caused error Server response could not be decoded using UTF-8
```
I have replaced some sensitive information with xxxxx.

### Configuration
**Operating System:**
Linux Debian10-1 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux

**Version of Python, python-telegram-bot & dependencies:**

``$ python -m telegram``

Output.
```
>>> python3 -m telegram
python-telegram-bot 11.1.0
certifi 2018.08.24
future 0.17.1
Python 3.7.3 (default, Apr  3 2019, 05:39:12)  [GCC 8.3.0]
```

So how to make the program ignore this unresolvable data without affecting other processing?

THX.

### Logs
Insert logs here (if necessary)

**Comment by Bibo-Joshi** [comment]
The raised `TelegramError` should not affect your bot, i.e. it should not shut down unexpectedly.
However, to actually understand, what kind of updates trigger those errors, you will need to provide a minimal working example.
A wild guess of mine is that it may have something to do with the `zh-hans` language code that appears in the update you posted.

**Comment by rikonaka** [comment]
I upgrade the `python-telegram-bot` to 'v12' then I also updated my bot's code, but this error still exists.
```
>>> python3 -m telegram
python-telegram-bot 12.2.0
certifi 2018.08.24
future 0.17.1
Python 3.7.3 (default, Apr  3 2019, 05:39:12)  [GCC 8.3.0]
```
I found a place that might cause this error to occur. In the first place where the error first appeared, there was a hexadecimal data appearing in callback_data.
```
"reply_markup": {
    "inline_keyboard": [
        [
            {
                "text": "\\u770b\\u4e0d\\u6e05\\uff0c\\u6362\\u4e00\\u4e2a",
                "callback_data":"\x9c\xffs\xb0:\xb4\xa6\xca\x96\xc2\\u0013\xd01\\u0004:\\u000f\x96x\xc1K\\u001b\x93#\xbe\xa7\\u001d#\xec\xc97\xff\x8e\xf8\x99\xcd2\xfcG\xe2x_\xd8\\r\xe2"
            }
        ],
        [
            {
                "text": "\\u4eba\\u5de5\\u901a\\u8fc7",
                "callback_data":"B\xb3\xc5\\u0002Q8\xc5\\"\xd8D%xq\xb4,\xd1v(\xf9! J\xbeh,
                0\x9fn\xad6\xeb\x9d\xe9\\u0013/\xa0FSp\xa6\xca\xf9\xe9g-"},
                            {
                    "text": "\\u4eba\\u5de5\\u62d2\\u7edd",
                    "callback_data":"\x9d%\x90\\u000f\\u000fD6@(S-hhw\xc0\xe2\xca\\u0001\\u0000\xdf\xd1\xe3\xd35\xa7\xab\xf88\\u0000\\u0016\xed+M0\xe5\x7f\xd4\xd6\'\xdc,$\x99\xd5\\f"
                }
            ]
        ]
    }
},
```
I am pretty sure that the inline_keyboard of this name has not been called in my program, and the data is not from my bot and my group, but it appears in the data that my bot accepts.

**Comment by rikonaka** [comment]
> The raised `TelegramError` should not affect your bot, i.e. it should not shut down unexpectedly.
> However, to actually understand, what kind of updates trigger those errors, you will need to provide a minimal working example.
> A wild guess of mine is that it may have something to do with the `zh-hans` language code that appears in the update you posted.

OK, here is some part of my code.
```
def main():
    updater = Updater(token=config.BOT_TOKEN, use_context=True)
    dp = updater.dispatcher
    def stop_and_restart():
        # Gracefully stop the Updater and replace the current process with a new one
        updater.stop()
        os.execl(sys.executable, sys.executable, *sys.argv)

    def restart(bot, update):
        update.message.reply_text('Bot is restarting...')
        Thread(target=stop_and_restart).start()

    job = updater.job_queue
    job.run_repeating(
        commands.xxxx, interval=6, first=0)
    dp.add_handler(MessageHandler(custom_filters.album,
                                  albums.album_collect, pass_job_queue=True), 1)
    dp.add_handler(MessageHandler(Filters.all, messages.message_process), 1)
    dp.add_handler(CommandHandler(
        'restart', restart, filters=Filters.user(username='@xxxx')))
    dp.add_handler(CommandHandler('xxxx', commands.xxxx), 2)
    dp.add_handler(CallbackQueryHandler(commands.command_query_back))

    # handle errors
    dp.add_error_handler(error)
    updater.start_polling()
    updater.idle()

if __name__ == '__main__':
    main()
```
I am guessing is the problem caused by python-telegram-bot error handling of some malicious hexadecimal data?

**Comment by Bibo-Joshi** [comment]
> Yes, the bot didn't shut down immediately, but it can not handle my command anymore, and repeating the error messages again and again.

That should not have happened, indeed.

Unfortunately your code snipped does not really help. Please provide an example that is runnable by simply inserting a bot token and also tell, what kind of update (e.g. text input) one needs to send in order to produce the error.

I'd like to add that's I'm far from being an expert on encoding ;)

**Comment by rikonaka** [comment]
> > Yes, the bot didn't shut down immediately, but it can not handle my command anymore, and repeating the error messages again and again.
> 
> That should not have happened, indeed.
> 
> Unfortunately your code snipped does not really help. Please provide an example that is runnable by simply inserting a bot token and also tell, what kind of update (e.g. text input) one needs to send in order to produce the error.
> 
> I'd like to add that's I'm far from being an expert on encoding ;)

Unfortunately, the code file is private and cannot be provided to you. If you are willing to provide me with an email address, I can provide you with all the logs in DEBUG mode, because this raw log file may involve some privacy that shouldn't display publicly on GitHub.

And then there is another thing to note that the update data that can't be processed has been increasing continuously, just like the exit of a queue is blocked.

I think a very simple way is there any unreliable way to discard this data, pretending that my bot has accepted it, instead of always causing errors due to this unhandled data,  this is a deadlock.

Just give me a way to discard this small probability data😄.

**Comment by Bibo-Joshi** [comment]
> Unfortunately, the code file is private and cannot be provided to you. If you are willing to provide me with an email address, I can provide you with all the logs in DEBUG mode, because this raw log file may involve some privacy that shouldn't display publicly on GitHub.

I'm not saying you should post the code of your bot. I'm saying you should provide a sample bot that produces a similar error.

> I think a very simple way is there any unreliable way to discard this data, pretending that my bot has accepted it, instead of always causing errors due to this unhandled data, this is a deadlock.

Well, you can always patch `telegram.utils.request._parse` to not raise an `TelegramError` on failed encoding, but that won't help getting to the source of the problem ;)

**Comment by rikonaka** [comment]
> I'm not saying you should post the code of your bot. I'm saying you should provide a sample bot that produces a similar error.
> 
> Well, you can always patch `telegram.utils.request._parse` to not raise an `TelegramError` on failed encoding, but that won't help getting to the source of the problem ;)

I don't quite understand what you mean.

If you want to test my bot to verify something, but now my bot can't handle any commands that cause the converted error, the message data can't be transferred from the `python-telegram-bot` module to my bot.

😂As I just said, the first data blocking in the `update` queue is the hexadecimal callback_query data with unknown effective. This is obviously a problem when the `python-telegram-bot` converts hexadecimal. Because of this problem, he was blocked in the first position of the `update` queue.

Thank you for your help.😋

**Comment by rikonaka** [comment]
> > Unfortunately, the code file is private and cannot be provided to you. If you are willing to provide me with an email address, I can provide you with all the logs in DEBUG mode, because this raw log file may involve some privacy that shouldn't display publicly on GitHub.
> 
> I'm not saying you should post the code of your bot. I'm saying you should provide a sample bot that produces a similar error.
> 
> > I think a very simple way is there any unreliable way to discard this data, pretending that my bot has accepted it, instead of always causing errors due to this unhandled data, this is a deadlock.
> 
> Well, you can always patch `telegram.utils.request._parse` to not raise an `TelegramError` on failed encoding, but that won't help getting to the source of the problem ;)

I try the the `telepot` module with this simple code, and I think I locate the error.
```
import telepot

token = '00000000000000000000000000000000'
TelegramBot = telepot.Bot(token)

print(TelegramBot.getUpdates())
```

Output.
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/telepot/api.py", line 133, in _parse
    text = response.data.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x9c in position 1435: invalid start byte
```
Maybe the `python-telegram-bot` support team can improve the relevant code of the conversion to deal with this small probability event.

**Comment by rikonaka** [comment]
I also get the 0x9c in the position 1435 in original data, which is the data in a `callback_query`.
```
"reply_markup": {
    "inline_keyboard": [
        [
            {
                "text": "\\u770b\\u4e0d\\u6e05\\uff0c\\u6362\\u4e00\\u4e2a",
------->      "callback_data":"\x9c\xffs\xb0:\xb4\xa6\xca\x96\xc2\\u0013\xd01\\u0004:\\u000f\x96x\xc1K\\u001b\x93#\xbe\xa7\\u001d#\xec\xc97\xff\x8e\xf8\x99\xcd2\xfcG\xe2x_\xd8\\r\xe2"
            }
        ],
        [
            {
                "text": "\\u4eba\\u5de5\\u901a\\u8fc7",
                "callback_data":"B\xb3\xc5\\u0002Q8\xc5\\"\xd8D%xq\xb4,\xd1v(\xf9! J\xbeh,
                0\x9fn\xad6\xeb\x9d\xe9\\u0013/\xa0FSp\xa6\xca\xf9\xe9g-"},
                            {
                    "text": "\\u4eba\\u5de5\\u62d2\\u7edd",
                    "callback_data":"\x9d%\x90\\u000f\\u000fD6@(S-hhw\xc0\xe2\xca\\u0001\\u0000\xdf\xd1\xe3\xd35\xa7\xab\xf88\\u0000\\u0016\xed+M0\xe5\x7f\xd4\xd6\'\xdc,$\x99\xd5\\f"
                }
            ]
        ]
    }
},
```

**Comment by Bibo-Joshi** [comment]
If also `telepot` gives an encoding error, my guess would be that the trouble comes from malformed updates, rather than `telepot`/`ptb` not being able to handle them. But again, I'm no expert on encoding and hence will be rather stepping back from this issue.

Just to clarify on the MWE: In order to investigate the issue (or any issue), in your case especially the bit about "The bot won't react to updates any more", we need to be able to recreate the problem you are facing. That is easier, if you provide a code snippet that produces the error. ;)

Sorry for not being helpful.

**Comment by rikonaka** [comment]
> If also `telepot` gives an encoding error, my guess would be that the trouble comes from malformed updates, rather than `telepot`/`ptb` not being able to handle them. But again, I'm no expert on encoding and hence will be rather stepping back from this issue.
> 
> Just to clarify on the MWE: In order to investigate the issue (or any issue), in your case especially the bit about "The bot won't react to updates any more", we need to be able to recreate the problem you are facing. That is easier, if you provide a code snippet that produces the error. ;)
> 
> Sorry for not being helpful.

However😅, I still think my code is not related to this problem.

As long as anyone forwards the message with this special `inline_keyboard` which contains the `0x9c` in data to any bot, all bots will cause deadlock like me because of the conversion data error.

You know, the key to the problem is that the `ptb` can't handle this `0x9c` in data correctly.

And the malicious message that someone else sent to my bot, I can't do any recurrence for you, sorry about that. If you need me I can provide this message in the log display.

I think the current priority should be to modify the conversion code of `ptb`, detect `0x9c` in the data, and then do the operation of discarding or rejecting `0x9c`.

**Comment by rikonaka** [comment]
> If also `telepot` gives an encoding error, my guess would be that the trouble comes from malformed updates, rather than `telepot`/`ptb` not being able to handle them. But again, I'm no expert on encoding and hence will be rather stepping back from this issue.
> 
> Just to clarify on the MWE: In order to investigate the issue (or any issue), in your case especially the bit about "The bot won't react to updates any more", we need to be able to recreate the problem you are facing. That is easier, if you provide a code snippet that produces the error. ;)
> 
> Sorry for not being helpful.

And, are you in the group of `ptb` in https://t.me/pythontelegrambotgroup or did you use the telegram?

I am already in this group, you can communicate with me in the group. This is more convenient if you use telegram.

**Comment by tsnoam** [comment]
Ok. It seems that this is a case where there's a non conforming client, sending non utf8 character.

Despite this is out of spec and shouldn't be supported, we can still try to be liberal and get something of the message.

Changing this line:
https://github.com/python-telegram-bot/python-telegram-bot/blob/10c9ec231305a023984b8f822b7674bdfe0aca04/telegram/utils/request.py#L171

To:
`...decode("utf-8", "backslashreplace")`
would probably work. 
Just need to verify compatibility with all supported python versions (should be just fine if I recall correctly).

**Comment by rikonaka** [comment]
Good try😅, but it still not fix the problem...
![image](https://user-images.githubusercontent.com/13602602/68963411-d5a44400-0811-11ea-8ef0-9c4a22f31993.png)

But there is new problem now.
```
2019-11-15 17:45:07,866 - telegram.ext.updater - ERROR - Error while getting Updates: Invalid server response
2019-11-15 17:45:07,870 - titswikibot.__main__ - WARNING - Update None caused error Invalid server response
2019-11-15 17:45:09,196 - telegram.ext.updater - ERROR - Error while getting Updates: Invalid server response
2019-11-15 17:45:09,203 - titswikibot.__main__ - WARNING - Update None caused error Invalid server response
2019-11-15 17:45:11,018 - telegram.ext.updater - ERROR - Error while getting Updates: Invalid server response
2019-11-15 17:45:11,020 - titswikibot.__main__ - WARNING - Update None caused error Invalid server response
2019-11-15 17:45:13,591 - telegram.ext.updater - ERROR - Error while getting Updates: Invalid server response
2019-11-15 17:45:13,591 - titswikibot.__main__ - WARNING - Update None caused error Invalid server response
^C2019-11-15 17:45:14,197 - telegram.ext.updater - INFO - Received signal 2 (SIGINT), stopping...
^C2019-11-15 17:45:16,694 - telegram.ext.updater - WARNING - Exiting immediately!
```

**Comment by tsnoam** [comment]
That means that the decoded message can't be parsed as json.
We have no interest in supporting non confirming clients.
Having said that, if you will be able to supply an example of such binary payload, we will try and see if we can tweak the code to be more liberal with what it accepts.

**Comment by rikonaka** [comment]
> That means that the decoded message can't be parsed as json.
> We have no interest in supporting non confirming clients.
> Having said that, if you will be able to supply an example of such binary payload, we will try and see if we can tweak the code to be more liberal with what it accepts.

No offense, are you sure you read all the comments?

I have provided the binary data payload which contains in the `update` JSON data that cannot be parsed.
```
"reply_markup": {
    "inline_keyboard": [
        [
            {
                "text": "\\u770b\\u4e0d\\u6e05\\uff0c\\u6362\\u4e00\\u4e2a",
------->        "callback_data":"\x9c\xffs\xb0:\xb4\xa6\xca\x96\xc2\\u0013\xd01\\u0004:\\u000f\x96x\xc1K\\u001b\x93#\xbe\xa7\\u001d#\xec\xc97\xff\x8e\xf8\x99\xcd2\xfcG\xe2x_\xd8\\r\xe2"
            }
        ],
        [
            {
                "text": "\\u4eba\\u5de5\\u901a\\u8fc7",
                "callback_data":"B\xb3\xc5\\u0002Q8\xc5\\"\xd8D%xq\xb4,\xd1v(\xf9! J\xbeh,
                0\x9fn\xad6\xeb\x9d\xe9\\u0013/\xa0FSp\xa6\xca\xf9\xe9g-"},
                            {
                    "text": "\\u4eba\\u5de5\\u62d2\\u7edd",
                    "callback_data":"\x9d%\x90\\u000f\\u000fD6@(S-hhw\xc0\xe2\xca\\u0001\\u0000\xdf\xd1\xe3\xd35\xa7\xab\xf88\\u0000\\u0016\xed+M0\xe5\x7f\xd4\xd6\'\xdc,$\x99\xd5\\f"
                }
            ]
        ]
    }
},
```
Is it clear enough now?
```
\x9c\xffs\xb0:\xb4\xa6\xca\x96\xc2\\u0013\xd01\\u0004:\\u000f\x96x\xc1K\\u001b\x93#\xbe\xa7\\u001d#\xec\xc97\xff\x8e\xf8\x99\xcd2\xfcG\xe2x_\xd8\\r\xe2
```

**Comment by tsnoam** [comment]
Can you please upload that as a binary file?

**Comment by rikonaka** [comment]
> Can you please upload that as a binary file?

I am pretty sure this is an `inline_keyboad` parameter data, not a binary file.

**Comment by tsnoam** [comment]
@rikonaka modify request.py and Instead of calling decode, do something like:
<code>open(&#x27;somefile&#x27;, &#x27;wb&#x27;).write(json_data)</code>
It will save the payload as binary data.

**Comment by rikonaka** [comment]
> open('somefile', 'wb').write(json_data)

Unfortunately, that json_data is not the problematic data just a moment ago, it has disappeared when it was tested.

**Comment by rikonaka** [comment]
[somefile.zip](https://github.com/python-telegram-bot/python-telegram-bot/files/3852760/somefile.zip)
OK, This data has magically appeared.

**Comment by tsnoam** [comment]
yeah. 
seems like `.decode('utf-8', 'replace')` is more appropriate than `backslashreplace` (which isn't supported in py2.7 anyhow, just checked).

**Comment by Poolitzer** [comment]
so, you wanna keep this open @tsnoam? Then we should change the label from wontfix to good first issue

**Comment by Bibo-Joshi** [comment]
Closed by #1623
